### PR TITLE
Basic Class Configuration Implemented

### PR DESCRIPTION
**Pull request by DarceyHaddow**
**Implemented the classes: Product, Quality, Report, CurrentOrder and Button**

1. Classes Product and Quality have been instantiated with objects in the main code. The main code has been refactored to make use of each classes functions while still maintaining the existing EPOS implementation. The price of each product now displays in the correct format: with two values after the decimal point for all prices.

_IMPACT_ new libraries have been imported in the Product class.

2. The other classes have been created but not yet implemented. I will be instantiated and extending these classes in later commits. At this point in the class configuration, other work should be able to be carried out in parallel to this.

_IMPACT_ The button class will need input from the engineer creating the GUI, and so it can be expanded at a later date.

_ADDITIONAL_ unrelated to the classes, I have refactored the functionality of the checkout and cancel buttons to use a function to set/reset the state of the report, this is to reduce code repetitions. This is in a function called setFrame().

### CODE REVIEW

**Review by howieros — APPROVED** [review]
*Ready for merge*
- does jbutton1 ln 399 of EPOSUI serve any purpose?
- some comments are unnecessary; could these be cut down?

File: src/my/epos/Button.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package my.epos;
+
+public class Button {
+    private int Height;
+    private int Width;
+    private String Colour;
+
+    private String Text;
+    //set up data type for an image?
```
Comment (amlopez3) [inline src/my/epos/Button.java:9]: Did you mean to leave this comment in? Seems more like a note to future self, would probably be better in Pull request comment to not clutter up code

File: src/my/epos/EposUI.java [hunk 2]
```diff
@@ -209,38 +209,59 @@ private void btnExitActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST
...
-    cmboProductPrice.addItem("Eclair: £0.70");
-    cmboProductPrice.addItem("Profiterole: £0.35");
-    cmboProductPrice.addItem("Mille Feuille: £1.40");
-    cmboProductPrice.addItem("Madeleine: £0.20");
-    cmboProductPrice.addItem("Cannele: £0.40");
-    cmboProductPrice.addItem("Macaron: £0.60");
-    cmboProductPrice.addItem("Pain au Chocolat: £0.85");
-    cmboProductPrice.addItem("Tarte Tartin: £4.60");
-    
+        //this populates the quantity combo box
```
Comment (amlopez3) [inline src/my/epos/EposUI.java:212]: Is there a way to populate the quantity combo box in a more dynamic way e.g using a loop? There seems to be a lot of repetition

File: src/my/epos/EposUI.java [hunk 3]
```diff
@@ -209,38 +209,59 @@ private void btnExitActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST
...
     private void btnCheckoutActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_btnCheckoutActionPerformed
-    
-    //this resets and clears the frame back to its original state
-    txtEndOfDayReport.setText("");
-    lblTotal.setText("");
-    cmboProductPrice.setSelectedIndex(0);
-    cmboQuantity.setSelectedIndex(0);
+
+        //this resets and clears the frame back to its original state
+        setFrame("","",0,0);
```
Comment (amlopez3) [inline src/my/epos/EposUI.java:264]: Nice refactoring :)

File: src/my/epos/Product.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package my.epos;
```
Comment (amlopez3) [inline src/my/epos/Product.java:1]: Nice, cohesive class

File: src/my/epos/Quantity.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package my.epos;
```
Comment (amlopez3) [inline src/my/epos/Quantity.java:1]: Nice, cohesive class. However, explanatory comments could maybe be added to external documentation e.g. README.md file

**Comment by DarceyHaddow** [comment]
> _Ready for merge_
> 
> * does jbutton1 ln 399 of EPOSUI serve any purpose?
> * some comments are unnecessary; could these be cut down?

Nice catch! The jButton1 doesn't serve any purpose: it was just an example of the button type. I'll add it as a workflow action to remove it for later commits 👍 

Same thing with the comments, I'll remove them in my next commit!

**Review by amlopez3 — APPROVED** [review]
Very nice refactoring :) Made comments on a few things to consider

File: src/my/epos/Quantity.java (hunk 5 above)
Comment (DarceyHaddow) [reply src/my/epos/Quantity.java:1]: Great idea!! I'll add a README file to cover all the comments I've put in to explain the classes :)

File: src/my/epos/Button.java (hunk 1 above)
Comment (DarceyHaddow) [reply src/my/epos/Button.java:9]: I'll add this to the README file too!

File: src/my/epos/EposUI.java (hunk 2 above)
Comment (howieros) [reply src/my/epos/EposUI.java:212]: This is just an intermediate step - we hope to store the data for product name and price in a csv file. From there, we will read that info in to populate the buttons / combo box

**Pull request merged**